element-connector/src/main/java/org/eclipse/californium/elements/Connector.java

```diff
@@ -91,4 +91,12 @@
 	 * @return the address
 	 */
 	public InetSocketAddress getAddress();
+
+	/**
+	 * Returns true if this is a TCP connector. TCP connector is subject to a different spec (CoAP-over-TCP) and
+	 * will requir CoAP endpoint to instantiate a different matcher, serializer and coap stack.
+	 *
+	 * @return true if this is a TCP based connector.
+     */
+	public boolean isTcp();
```

Comment:
I'd rather have a method checking support for a particular scheme (e.g. _coap_ or _coap+tcp_). WDYT?